pkg/operator/staticpod/controller/monitoring/monitoring_resource_controller.go

```diff
@@ -106,7 +106,7 @@ func (c MonitoringResourceController) sync() error {
 		return nil
 	}
 
-	directResourceResults := resourceapply.ApplyDirectly(c.kubeClient, c.eventRecorder, c.mustTemplateAsset,
+	directResourceResults := resourceapply.ApplyDirectly((&resourceapply.ClientHolder{}).WithKubernetes(c.kubeClient), c.eventRecorder, c.mustTemplateAsset,
```

Comment:
little ugly, isn't it? Not that I have a better idea.

Reply:
Using a generic client using the rest mapper would help. It's what controller-runtime does. Maybe just use their client?

Reply:
> Using a generic client using the rest mapper would help. It's what controller-runtime does. Maybe just use their client?

that loses all type safety guarantees.

Reply:
and at this point, I'd like to get ourselves on sane footing for resources before we do surgery on the authentication operator.